### PR TITLE
ci: refactor `build-channel` and add documentation

### LINKED ISSUE #390: docs: describe usage of build-channel
The usage of build-channel should be clearly documented to allow anyone to be able to update channels, just like how `components.toml` is documented.

This should be added to the developer guide: https://fuellabs.github.io/fuelup/v0.18.0/developer_guide/index.html

### PR DESCRIPTION
**Pull request by bingcicle**
closes #390 

This PR documents `build-channel` within the fuelup developer guide.

This PR also makes the arguments/options more ergonomic - previously  we had to declare either `latest` or `nightly` to build a channel, and to build a `beta-3` channel I had to use `latest` as an arg (even though it doesn't make sense).

This PR:

1) makes building `nightly` into a flag, indicated by `--nightly`
2) makes including `github_run_id` an option, because manually running this script does not generate an id (which is done in the GitHub CI)

### CODE REVIEW

**Comment by bingcicle** [comment]
Realized I did not request for a review here, oops.

**Comment by bingcicle** [comment]
blocked by https://github.com/FuelLabs/fuelup/pull/394

File: docs/src/developer_guide/index.md [hunk 1]
```diff
@@ -8,3 +8,4 @@ This section will go through:
 
 - How to setup `fuelup` in your environment for contributions,
 - How to add support for a component to fuelup
+- How 'build-channel' is used in the CI, and how to manually use it to build a channel for fuelup
```
Comment (kayagokalp) [inline docs/src/developer_guide/index.md:11]:
```suggestion
- How `build-channel` is used in the CI, and how to manually use it to build a channel for fuelup
```

Should this be like this or is `'` intentional?

**Review by kayagokalp — COMMENTED** [review]
lgtm, just left a tiny nit/question

File: docs/src/developer_guide/building_a_channel.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+`build-channel` is a Rust script that creates a [channel] that serves as a source of distribution of
+the Fuel toolchain. This is accomplished by querying and collecting a list of downloadable components that
+Fuel Labs publishes, creating a TOML file based on the collated data, which is then consumed by `fuelup`
+during usage.
+
+To learn about the args and options used in the script, skip to [Usage].
+
+## Use cases
+
+There are 2 main ways where the `build-channel` script is used: in the CI, and manually.
```
Comment (mitchmindtree) [inline docs/src/developer_guide/building_a_channel.md:12]:
```suggestion
There are 2 main ways the `build-channel` script is used: in the CI, and manually.
```

**Review by mitchmindtree — APPROVED** [review]
Awesome :ok_hand:

**Pull request merged**